### PR TITLE
Map constructor throws NPE with null keys

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
Instead it should give a more meaningful error message.

```
presto:nyigitbasi> select map(array[null], array[null]);
Query 20150109_064055_00000_c84kn failed: null
java.lang.NullPointerException
        at com.facebook.presto.operator.scalar.MapConstructor.specialize(MapConstructor.java:86)
        at com.facebook.presto.metadata.FunctionRegistry.resolveFunction(FunctionRegistry.java:360)
        
presto:nyigitbasi> select map(array[null], array[1]);
Query 20150109_064057_00001_c84kn failed: null
java.lang.NullPointerException
        at com.facebook.presto.operator.scalar.MapConstructor.specialize(MapConstructor.java:86)
        at com.facebook.presto.metadata.FunctionRegistry.resolveFunction(FunctionRegistry.java:360)
```

### CODE REVIEW

**Comment by nezihyigitbasi** [comment]
@cberner Rebased & updated the PR to handle another edge case including null keys.

File: presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java [hunk 1]
```diff
@@ -146,6 +146,9 @@ public void testJsonToMap()
     public void testSubscript()
             throws Exception
     {
+        assertInvalidFunction("MAP(ARRAY [null], ARRAY [1])", "map key cannot be null");
```
Comment (cberner) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: This is the wrong message to be checking for. The function shouldn't even resolve, since array[null] is of type array<unknown> and is therefore not comparable. Looks like there's a bug in the signature code that I'm fixing, where this is matching

**Comment by cberner** [comment]
We need to merge https://github.com/facebook/presto/pull/2276 before this one to avoid the TODO in MapConstructor

**Comment by cberner** [comment]
Merged. Thanks!
